src/SayMore/UI/ComponentEditors/BindingHelper.cs

```diff
@@ -187,11 +187,59 @@ private void BindControl(Control ctrl)
 			else
 			{
 				ctrl.Validating += HandleValidatingControl;
+				ValidateOnHide(ctrl);
 				if (ctrl == _componentFileIdControl)
 					ctrl.KeyPress += HandleIdFieldKeyPress;
 			}
 		}
 
+		HashSet<Control> _validateOnHideControls = new HashSet<Control>();
+
+		private bool _runningValidations;
+		// If the control or any of its parents gets hidden, e.g., by switching to another tab,
+		// perform validation, even if the new tab does not assign focus to something, which
+		// would normally trigger it. This ensures some important side effects of validation
+		// don't get skipped (e.g., updating a Person's name in Contributors).
+		void ValidateOnHide(Control start)
```

Comment:
The only reference to this new method is inside itself. Doesn't seem like it will ever get called.